rita_common/src/peer_listener/mod.rs

```diff
@@ -156,52 +51,56 @@ fn listen_to_available_ifaces(peer_listener: &mut PeerListener) {
 
     info!(
         "PEER LISTENER: Done listening, Our interface list looks like : {:?}",
-        peer_listener.interfaces
+        get_interfaces()
     );
 }
 
-/// Returns a copy of PL lazy static var
-fn get_pl_copy() -> PeerListener {
-    PEER_LISTENER.read().unwrap().clone()
-}
-
-/// Sets the PeerListener lazy static to the given value
-fn set_pl(pl: PeerListener) {
-    *PEER_LISTENER.write().unwrap() = pl;
-}
-
 /// Ticks the peer listener module sending ImHere messages and receiving Hello messages from all
 /// peers over UDP
 pub fn peerlistener_tick() -> PeerListener {
     trace!("Starting PeerListener tick!");
 
-    let mut pl = get_pl_copy();
+    let pl = match get_pl_copy() {
+        Some(pl) => pl,
+        // we should never get here in prod as the default is created in the initialization, but this is a failsafe either way...
```

Comment:
Why not move this check into get_pl_copy that way it's not in the main program logic.